apps/download-service/src/app/modules/regulation-documents/regulation-documents.controller.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  Body,
+  Controller,
+  Header,
+  Inject,
+  Post,
+  Res,
+  Param,
+  UseGuards,
+} from '@nestjs/common'
+import { ApiOkResponse } from '@nestjs/swagger'
+import { Response } from 'express'
+import { RegulationsAdminClientService } from '@island.is/clients/regulations-admin'
+import { RegulationsService } from '@island.is/clients/regulations'
+import { AdminPortalScope } from '@island.is/auth/scopes'
+import type { User } from '@island.is/auth-nest-tools'
+import {
+  CurrentUser,
+  IdsUserGuard,
+  Scopes,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
+import { GetRegulationDraftDocumentDto } from './dto/getRegulationDraftDocument.dto'
+import {
+  RegulationDraft,
+  RegulationPdfInput,
+} from '@island.is/regulations/admin'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Scopes(
+  AdminPortalScope.regulationAdmin,
+  AdminPortalScope.regulationAdminManage,
+)
+@Controller('regulation')
+export class RegulationDocumentsController {
+  constructor(
+    @Inject(RegulationsService)
+    private readonly regulationService: RegulationsService,
+    @Inject(RegulationsAdminClientService)
+    private regulationsAdminClientService: RegulationsAdminClientService,
+  ) {}
+
+  @Post('/draft/:regulationId')
+  @Header('Content-Type', 'application/pdf')
+  @ApiOkResponse({
+    content: { 'application/pdf': {} },
+    description: 'Get a PDF document for a draft regulation',
+  })
+  async getDraftRegulationPdf(
+    @Param('regulationId') regulationId: string,
+    @CurrentUser() user: User,
+    @Body() resource: GetRegulationDraftDocumentDto,
```

Comment:
Can you use `user.authorization` instead? Shouldn't need to define GetRegulationDraftDocumentDto just to get the access token.

Reply:
Oh! Probably a bug in the auth library. Nevermind then.

Reply:
Yes, that would be great, as we are doing the same thing for the finance download service. Would be a nice change. 🙌